src/service/data_products/common_functions.py

```diff
@@ -352,6 +357,10 @@ async def query_table(
         raise errors.IllegalParameterError(
                 f"No such field for collection {filters.collection_id} load version "
                 + f"{filters.load_ver}: {filters.sort_on}")
+    if not filters.selection_spec.is_null_subset():
+        fields = [names.FLD_SELECTED_SAFE] + fields
```

Comment:
it's ok to have duplicate `names.FLD_SELECTED_SAFE`/ `names.FLD_MATCHED_SAFE` in `fields` array?

Reply:
I'm not understanding, those aren't duplicate names

Reply:
can `fields` array already have `names.FLD_SELECTED_SAFE`? unlikely?

Reply:
They don't actually exist in the database, so unless the calling code does something really weird, no.

Reply:
Currently the field names are coming from the column specs, which don't have those names.

Reply:
👍